internal/controller/vrg_volrep.go

```diff
@@ -937,6 +944,106 @@ func (v *VRGInstance) reconcileMissingVR(pvc *corev1.PersistentVolumeClaim, log
 	return vrMissing, !requeue
 }
 
+func (v *VRGInstance) buildVolumeGroupReplicationName(pvc *corev1.PersistentVolumeClaim) (string, error) {
+	storageID, ok := pvc.GetLabels()[ConsistencyGroupLabel]
+	if !ok {
+		v.log.Info("Missing storageID for PVC %s/%s", pvc.GetNamespace(), pvc.GetName())
+
+		return "", fmt.Errorf("missing storageID for PVC %s/%s", pvc.GetNamespace(), pvc.GetName())
+	}
+	
+	vgrName := storageID + "-vgr"
```

Comment:
There can be multiple workloads independently DR protected in the same namespace. These can have PVCs from the same SC with the same StorageID. The name will then conflict.

I would suggest, munging this with the VRG name as well. We cannot have 2 VRGs with the same name in the same namespace, so using that as part of the name maybe an option. It does create a possibility that we will have a longer name, so any shortening here may assist in that.

Reply:
I changed VGR name to {vrg}-{storageID}, @ShyamsundarR what is we will remove -drpc from vrg name? It will shorten the whole name a little